src/NServiceBus.RabbitMQ/Routing/DefaultRoutingKeyConvention.cs

```diff
@@ -21,6 +22,12 @@ static string GetRoutingKey(Type type, string key = "")
             var interfaces = type.GetInterfaces()
                 .Where(i => !IsSystemType(i) && !IsNServiceBusMarkerInterface(i)).ToList();
 
+            if (interfaces.Count > 1)
+            {
+                Logger.WarnFormat(@"The direct routing topology may not deliver message types with interfaces in their inheritance hierarchy. Refrain from using interface
```

Comment:
I wonder if this message should borrow more from the language used in the warning note under https://docs.particular.net/nservicebus/rabbitmq/routing-topology#direct-routing-topology-enabling-direct-routing-topology